File: grida-apis/apis-security/src/main/java/org/grida/config/JwtProperties.java
package org.grida.config;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;

@Getter
@RequiredArgsConstructor
@ConstructorBinding
@ConfigurationProperties(prefix = "jwt")
public class JwtProperties {

    private final String secretKey;
    private final long accessTokenValidityInSecond;
    private final long refreshTokenValidityInSecond;
}

File: grida-core/core-api/src/main/kotlin/org/grida/config/JwtProperties.kt
package org.grida.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties("jwt")
data class JwtProperties(
    val secretKey: String,
    val accessTokenExpired: Long,
    val refreshTokenExpired: Long,
)
